sftp_lambda_integration/regenstrief_s3_sftp.py

```diff
@@ -11,13 +13,54 @@
 logger.setLevel(os.getenv("LOGGING_LEVEL", "INFO"))
 
 
+def get_rsa_private_key_from_secrets(
+    secret_name="sftp-ri-rsa-key", region_name="us-east-1"
+):
+    """
+    Retrieve a specific RSA private key from AWS Secrets Manager.
+
+    :param secret_name: Name of the secret in AWS Secrets Manager
+    :param region_name: AWS region where the secret is stored
+    :return: RSA private key string
+    """
+    try:
+        # Create a Secrets Manager client
+        session = boto3.session.Session()
+        client = session.client(service_name="secretsmanager", region_name=region_name)
+        secret_string = client.get_secret_value(SecretId=secret_name)["SecretString"]
+        try:
```

Comment:
instead of the 2 nested try blocks, could we do this?
```
try:
  client.get_secret_value
except botocore.exceptions.ClientError as e:
  [...]
try:
  parsed_secret =
except json.JSONDecodeError:
  [...]
```